### PR TITLE
fix: split button overflow

### PR DESCRIPTION
**Pull request by stefanoScalzo**
## Related Issue
Closes  https://github.com/SAP/fundamental-styles/issues/1721

## Description
Add text overflow to the text of split button
For regular cosy button 9.75rem is used because the other button takes 2.25rem and the max-width is always 12rem
For compact 10rem is used to keep this same 12rem max-width

### Before:
image

### After:
Screen Shot 2020-12-15 at 1 16 52 PM

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/button-split.scss [hunk 1]
```diff
@@ -76,4 +76,16 @@ $fd-button-split-margin: 0.125rem / 2;
       }
     }
   }
+
+  &__text {
```
Comment (InnaAtanasova) [inline src/button-split.scss:80]: Can't you target the first child instead of adding an additional modifier class?

File: src/button-split.scss [hunk 2]
```diff
@@ -76,4 +76,16 @@ $fd-button-split-margin: 0.125rem / 2;
...
+    max-width: 9.75rem;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
```
Comment (InnaAtanasova) [inline src/button-split.scss:84]: There's a mixin for this :D

File: src/button-split.scss (hunk 1 above)
Comment (salarenko) [reply src/button-split.scss:80]: But text is not wrapped in any element, and you cannot really target a `TextNode`.

**Review by salarenko — COMMENTED** [review]
Where does the max-width come from? I don't see any requirements in the specs.
Shouldn't standard button component cover such cases?

**Comment by stefanoScalzo** [comment]
> Where does the max-width come from? I don't see any requirements in the specs.
> Shouldn't standard button component cover such cases?

Screen Shot 2020-12-18 at 9 29 20 AM

File: stories/button/button.stories.js [hunk 3]
```diff
@@ -284,7 +284,7 @@ menuButton.parameters = {
 
 export const splitMenuButton = () => `
 <div class="fd-button-split fd-has-margin-right-small" role="group" aria-label="button-split">
-  <button class="fd-button" aria-label="button">Button with text</button>
+  <button class="fd-button fd-button-split__text" aria-label="button">Button with a lot of text</button>
```
Comment (JKMarkowski) [inline stories/button/button.stories.js:287]: Could you just wrap the text inside `<span class="fd-button__text">Button with a lot of text</span>`?
And inside split-button just add `fd-button__text { max-width: ... }`

Comment (stefanoScalzo) [reply stories/button/button.stories.js]: fixed

File: src/button-split.scss [hunk 4]
```diff
@@ -76,4 +76,15 @@ $fd-button-split-margin: 0.125rem / 2;
       }
     }
   }
+
+  .fd-button__text {
```
Comment (JKMarkowski) [inline src/button-split.scss:80]: Let's add `fd-namespace` instead of `fd-`

Comment (stefanoScalzo) [reply src/button-split.scss]: fixed

**Review by JKMarkowski — APPROVED** [review]
LGTM

File: src/button-split.scss [hunk 5]
```diff
@@ -76,4 +76,15 @@ $fd-button-split-margin: 0.125rem / 2;
       }
     }
   }
+
+  .#{$fd-namespace}-button__text {
+    @include fd-ellipsis();
```
Comment (InnaAtanasova) [inline src/button-split.scss:81]: Shouldn't this have a reset?

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
I think it's a Breaking change and should be documented

File: src/button-split.scss (hunk 5 above)
Comment (InnaAtanasova) [reply src/button-split.scss:81]: Ok, I understand what you did. You don't need the reset coz it's `.fd-button__text`, but if this is the case, you can't just use it here. Split button is a separate component from Button and this violates the self-contained rule. You will need a modifier class for these rules

Comment (stefanoScalzo) [reply src/button-split.scss:138]: fixed

**Comment by InnaAtanasova** [comment]
Tests are failing

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**